src/texture/png.rs

```diff
@@ -0,0 +1,33 @@
+use std::io::Cursor;
+
+use image::codecs::png::PngDecoder;
+use image::{ColorType, DynamicImage, ImageBuffer, ImageDecoder};
+
+use super::Texture;
+
+/// Decodes a PNG image into an RGBA texture.
+pub fn decode(png: &[u8]) -> Texture {
+    let decoder = PngDecoder::new(Cursor::new(png)).unwrap();
+    let (width, height) = decoder.dimensions();
+    let color_type = decoder.color_type();
+
+    let mut data = vec![0_u8; decoder.total_bytes() as usize];
```

Comment:
```suggestion
    let mut data: Vec<u8> = Vec::with_capacity(decoder.total_bytes() as usize);
```

Reply:
That won't work, `PngDecoder::read_image` takes a slice and checks its length to read the PNG into it.

Reply:
well create a slice out of a Vec?

Reply:
The slice will be of length 0.